quickwit/quickwit-search/src/root.rs

```diff
@@ -1103,6 +1121,104 @@ pub async fn root_search(
+    };
+    let indexes_metadata: Vec<IndexMetadata> = metastore
+        .list_indexes_metadata(list_indexes_metadatas_request)
+        .await?
+        .deserialize_indexes_metadata()
+        .await?;
+
+    check_all_index_metadata_found(&indexes_metadata[..], &search_request.index_id_patterns[..])?;
+    if indexes_metadata.is_empty() {
+        return Ok(SearchPlanResponse {
+            result: serde_json::to_string(&SearchPlanResponseRest {
+                quickwit_ast: QueryAst::MatchAll,
+                tantivy_ast: String::new(),
+                searched_splits: Vec::new(),
+                storage_requests: StorageRequestCount::default(),
+            })?,
+        });
+    }
+    let doc_mapper = build_doc_mapper(
+        &indexes_metadata[0].index_config.doc_mapping,
+        &indexes_metadata[0].index_config.search_settings,
+    )
+    .map_err(|err| SearchError::Internal(format!("failed to build doc mapper. cause: {err}")))?;
+
+    let request_metadata = validate_request_and_build_metadata(&indexes_metadata, &search_request)?;
+    let split_metadatas = refine_and_list_matches(
+        &mut metastore,
+        &mut search_request,
+        indexes_metadata,
+        request_metadata.query_ast_resolved.clone(),
+        request_metadata.sort_fields_is_datetime,
+        request_metadata.timestamp_field_opt,
+    )
+    .await?;
+
+    let (query, mut warmup_info) = doc_mapper.query(
+        doc_mapper.schema(),
+        &request_metadata.query_ast_resolved,
+        true,
+    )?;
+    let merge_collector = make_merge_collector(&search_request, &Default::default())?;
+    warmup_info.merge(merge_collector.warmup_info());
+    warmup_info.simplify();
+
+    let split_ids = split_metadatas
+        .into_iter()
+        .map(|split| format!("{}/{}", split.index_uid.index_id, split.split_id))
+        .collect();
+    // this is an upper bound, we'd need access to a hotdir for more precise results
+    let fieldnorm_query_count = if warmup_info.field_norms {
+        doc_mapper
+            .schema()
+            .fields()
+            .filter(|(_, entry)| entry.has_fieldnorms())
+            .count()
+    } else {
+        0
+    };
+    let sstable_query_count = warmup_info.term_dict_fields.len()
+        + warmup_info
+            .terms_grouped_by_field
+            .values()
+            .map(|terms| terms.len())
+            .sum::<usize>();
+    let position_query_count = warmup_info
+        .terms_grouped_by_field
+        .values()
+        .map(|terms| {
+            terms
+                .values()
+                .filter(|load_position| **load_position)
+                .count()
+        })
+        .sum();
+    Ok(SearchPlanResponse {
+        result: serde_json::to_string(&SearchPlanResponseRest {
+            quickwit_ast: request_metadata.query_ast_resolved,
+            tantivy_ast: format!("{query:#?}"),
+            searched_splits: split_ids,
+            storage_requests: StorageRequestCount {
```

Comment:
that's cool you extended the thing to a warmup info etc! can we actually get the number of bytes as well?

Reply:
that's something i would find interesting too, but right now it's not as easy to get. All metadata we get here, we get only through the metastore, with zero call to the storage. To get more, we would need to actually download things from storage, which starts having more of a `EXPLAIN ANALYSE` than an `EXPLAIN` semantic. I think that should be left out of scope of this PR, but let's open a ticket so eventually people can pass `?analyze=true` or something to get that information as well